### PR TITLE
Modified Get-All_SDS_Attributes_NoRosters.ps1 script to have GraphAPI

### PR DESCRIPTION
**Pull request by ddwivedy84**
### Related Issue
https://dev.azure.com/1edu/DIP%20Master/_queries/edit/132826/
### Summary
Modified script to have GraphAPI instead of ADAL cmdlets

### Reason
Old scripts some functionality was not working 

### Proposed Change(s)
ADAL cmdlets to GraphAPI

### Details
_Detailed description of the change(s)_

### Additional Info
_Any additional information or context_

### References
[Example](www.microsoft.com)

### Checklist
- [ ] Documentation

### CODE REVIEW

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 1]
```diff
@@ -1,201 +1,94 @@
 ﻿<#
-Script Name:
-Get-All_SDS_Attributes_No_Rosters.ps1
+.SYNOPSIS
+    This script is designed to get all SDS objects and attributes, except for Rostering. The result is only 4 CSV files in a trimmed down version of the standard SDS format, as detailed on the page https://aka.ms/sdscsvattributes. The 4 CSV files generated are School.CSV, Section.CSV, Student.CSV, and Teacher.CSV. The 2 files not generated are StudentEnrollment.CSV and TeacherRoster.CSV. This reduces the run overall time of the SDS attributes script by about 80%. If you need all SDS attributes run the Get-All_SDS_Attributes.ps1 instead.
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:3]: Remove spacing.

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 2]
```diff
@@ -214,15 +107,31 @@ function Get-SdsSchools
...
+
     foreach($au in $list)
     {
+        #SIS ID,Name,School Number,School NCES_ID,State ID,Grade Low,Grade High,Principal SIS ID,Principal Name,Principal Secondary Email,Address,City,State,Zip,Phone,Zone,Country
         $data += [pscustomobject]@{
             "SIS ID" = $au.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_SchoolId
             "Name" = $au.DisplayName
             "School Number" = $au.extension_fe2174665583431c953114ff7268b7b3_Education_SchoolNumber 
             "School NCES_ID" = $au.extension_fe2174665583431c953114ff7268b7b3_Education_SchoolNationalCenterForEducationStatisticsId
+            "State ID" = $au.extension_fe2174665583431c953114ff7268b7b3_Education_StateId
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:119]: These fields were not in orginal script.  I would leave them out.

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 3]
```diff
@@ -327,40 +234,22 @@ function Get-Groups
...
         }
     }
-    While ($responseObject.odatanextLink -ne $null)
-
     return $list
 }
 
-
-
 function Get-SdsSection
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:253]: Get-SdsSection and Get-Section don't look updated

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 4]
```diff
@@ -1,201 +1,94 @@
...
+1. Install Microsoft Graph Powershell Module with command 'Install-Module Microsoft.Graph'.
+PowerShell 7 and later is the recommended PowerShell version for use with the Microsoft Graph PowerShell SDK on all platforms (https://docs.microsoft.com/en-us/graph/powershell/installation#supported-powershell-versions).
 
-2. Install the AAD PowerShell Module from http://msdn.microsoft.com/en-us/library/azure/jj151815.aspx#bkmk_installmodule
+2. Make sure to download common.ps1 to the same folder of the script which has common functions needed.  https://github.com/OfficeDev/O365-EDU-Tools/blob/master/SDS%20Scripts/common.ps1
+Command to download the function: Invoke-WebRequest -Uri "https://raw.githubusercontent.com/OfficeDev/O365-EDU-Tools/master/SDS%20Scripts/common.ps1" -OutFile ".\common.ps1" -ErrorAction Stop -Verbose
 
 3. Check that you can connect to your tenant directory from the PowerShell module to make sure everything is set up correctly.
 
     a. Open a separate PowerShell session
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:30]: NIT: Remove spaces before new lines codes

**Review by AyronJohnson — CHANGES_REQUESTED** [review]
See comments.  Highlights are a few functions not using send-webrequest and old uri (they don't look updated) and extra fields not inlcuded in original script.  Otherwise minor fixes.

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 (hunk 2 above)
Comment (ddwivedy84) [reply SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1]: Updated to have fields that were in the original script.

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 5]
```diff
@@ -1,201 +1,94 @@
...
+        throw "Unable to download common.ps1"
+    }
 }
 
+#import file with common functions
+. .\common.ps1
+
 function Export-SdsSchools
 {
     $fileName = $eduObjSchool.ToLower() + $(if ($AppendTenantIdToFileName) { "-" + $authToken.TenantId } else { "" }) +".csv"
```
[the remark this answers is not in the document]
Comment (ddwivedy84) [reply SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1]: Updated as suggested

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 6]
```diff
@@ -510,130 +336,69 @@ function Get-Users
...
 {
     $graphEndPoint = $GraphEndpointPPE
-    $authEndPoint = $AuthEndpointPPE
 }
 
 $activityName = "Reading SDS objects in the directory"
 
+$graphscopes = "User.Read.All, GroupMember.Read.All, Group.Read.All, Directory.Read.All, AdministrativeUnit.Read.All"
+
+$tenantInfo = Get-MgOrganization
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:366]: I think you need to call these lines after initialize

Comment (ddwivedy84) [reply SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1]: done

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 7]
```diff
@@ -430,17 +255,18 @@ function Get-SdsTeachers
...
+            "DisplayName" = $user.DisplayName  
             "UserPrincipalName" = $user.userPrincipalName          
-	    "SIS ID" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_TeacherId
+	        "SIS ID" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_TeacherId
             "School SIS ID" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_SchoolId
             "Teacher Number" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_TeacherNumber
             "Status" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_TeacherStatus
             "Secondary Email" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_Email
-	    "ObjectID" = $user.objectID
+	        "ObjectID" = $user.objectID
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:269]: This is not populating

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 [hunk 8]
```diff
@@ -479,17 +304,18 @@ function Get-SdsStudents
...
-	    "SIS ID" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_StudentId
+            "DisplayName" = $user.displayname
+	        "UserPrincipalName" = $user.userPrincipalName        
+	        "SIS ID" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_StudentId
             "School SIS ID" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_SyncSource_SchoolId
             "Student Number" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_StudentNumber
             "Status" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_StudentStatus
             "Secondary Email" = $user.extension_fe2174665583431c953114ff7268b7b3_Education_Email
-	    "ObjectID" = $user.ObjectID
+	        "ObjectID" = $user.ObjectID
```
Comment (AyronJohnson) [inline SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1:318]: Not populating.  Maybe change to $user.Id

File: SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1 (hunk 7 above)
Comment (ddwivedy84) [reply SDS Scripts/Get-All_SDS_Attributes_NoRosters.ps1]: Changed from ObjectID to id. Its populating now.please check

**Pull request merged**